pkg/operator/rules.go

```diff
@@ -142,7 +162,8 @@ func (pr *PrometheusRuleSelector) Select(namespaces []string) (map[string]string
 			continue
 		}
 
-		content, err = generateRulesConfiguration(pr.ruleFormat, promRule.Spec, pr.logger)
+		pr.logger = log.With(pr.logger, "prometheusrule", promRule.Name, "prometheusrule-namespace", promRule.Namespace)
```

Comment:
this would overwrite the logger field at every loop. See my other comment above to make generateRulesConfiguration() a pointer receiver's method.